src/app/common/stores/applications/SsvWeb/Validator.store.ts

```diff
@@ -178,7 +181,101 @@ class ValidatorStore extends BaseStore {
+        store.dispatch(setIsLoading(false));
+        notificationsStore.showMessage(e.message, 'error');
+        store.dispatch(setIsShowTxPendingPopup(false));
+        resolve(false);
+      }
+    });
+  }
+
+  /**
+   * Bulk remove validators
+   */
+  async bulkRemoveValidators(validators: string[], operatorIds: number[]): Promise<boolean> {
+    const clusterStore: ClusterStore = this.getStore('Cluster');
+    // const applicationStore: ApplicationStore = this.getStore('Application');
+    const notificationsStore: NotificationsStore = this.getStore('Notifications');
+    const contract = getContractByName(EContractName.SETTER);
+    store.dispatch(setIsLoading(true));
+    const myAccountStore: MyAccountStore = this.getStore('MyAccount');
+    const clusterData = await clusterStore.getClusterData(clusterStore.getClusterHash(operatorIds));
+    return new Promise(async (resolve) => {
+      try {
+        const tx = await contract.bulkRemoveValidator(validators, operatorIds, clusterData);
+        if (tx.hash) {
+          store.dispatch(setTxHash(tx.hash));
+          store.dispatch(setIsShowTxPendingPopup(true));
+        }
+        const receipt = await tx.wait();
+        if (receipt.blockHash) {
+          ApiParams.initStorage(true);
+          await executeAfterEvent(async () => !!await ContractEventGetter.getInstance().getEventByTxHash(receipt.transactionHash), async () => this.refreshOperatorsAndClusters(resolve, true), myAccountStore.delay);
+        }
+      } catch (e: any) {
+        store.dispatch(setIsLoading(false));
+        notificationsStore.showMessage(e.message, 'error');
+        store.dispatch(setIsShowTxPendingPopup(false));
+        resolve(false);
+      }
+    });
+  }
+
+  /**
+   * Exit validator
+   */
+  async exitValidator(publicKey: string, operatorIds: number[]): Promise<boolean> {
+    // const applicationStore: ApplicationStore = this.getStore('Application');
+    const notificationsStore: NotificationsStore = this.getStore('Notifications');
+    const contract = getContractByName(EContractName.SETTER);
+    store.dispatch(setIsLoading(true));
+
+    return new Promise(async (resolve) => {
+      try {
+        const tx = await contract.exitValidator(publicKey, operatorIds);
+        if (tx.hash) {
+          store.dispatch(setTxHash(tx.hash));
+          store.dispatch(setIsShowTxPendingPopup(true));
+        }
+        const receipt = await tx.wait();
+        if (receipt.blockHash) {
+          store.dispatch(setIsShowTxPendingPopup(false));
+          store.dispatch(setIsLoading(false));
+          resolve(true);
+        }
+      } catch (e: any) {
+        store.dispatch(setIsLoading(false));
+        notificationsStore.showMessage(e.message, 'error');
+        store.dispatch(setIsShowTxPendingPopup(false));
+        resolve(false);
+      }
+    });
+  }
+
+  /**
+   * Bulk exit validator
+   */
+  async bulkExitValidators(validators: string[], operatorIds: number[]): Promise<boolean> {
```

Comment:
We have a lot of duplications here, not just the 'if (tx.hash)' We should extract functionality from Validator.store.ts into validator service and have it be extended by bulk or single. @IlyaVi can we do it in the context of this PR?

Reply:
All transactions have the same structure, I want to create some wrapper for this but let's make it in another PR to note block this one. 
Let's leave for now all comments about duplication